src/models/run_group_is_from_github.rs

```diff
@@ -0,0 +1,873 @@
+pub struct NewRunGroupIsFromGithub {
+    pub run_group_id: Uuid,
+    pub repo: String,
+    pub owner: String,
+    pub issue_number: i32,
+    pub author: String,
+    pub base_commit: String,
+    pub head_commit: String,
+    pub test_input_key: Option<String>,
+    pub eval_input_key: Option<String>,
+}
+
+impl RunGroupIsFromGithubData {
+    /// Queries the DB for a run_group_is_from_github with the specified run_group_id
+    ///
+    /// Queries the DB using `conn` to retrieve the first row with a run_group_id value of `id`
+    /// Returns a result containing either the retrieved run_group_is_from_github as a
+    /// RunGroupIsFromGithubData instance or an error if the query fails for some reason or if no
+    /// run_group_is_from_github is found matching the criteria
+    ///
+    /// This is here for api completeness
+    #[allow(dead_code)]
+    pub fn find_by_run_group_id(
+        conn: &PgConnection,
+        id: Uuid,
+    ) -> Result<Self, diesel::result::Error> {
+        run_group_is_from_github
+            .filter(run_group_id.eq(id))
+            .first::<Self>(conn)
+    }
+
+    /// Queries the DB for a run_group_is_from_github for the specified run_id
+    ///
+    /// Queries the DB using `conn` to retrieve the first row with a run_group_id that corresponds
+    /// to a run_group record to which the run with id `run_id` belongs
+    /// Returns a result containing either the retrieved run_group_is_from_github as a
+    /// RunGroupIsFromGithubData instance or an error if the query fails for some reason or if no
+    /// run_group_is_from_github is found matching the criteria
+    pub fn find_by_run_id(
+        conn: &PgConnection,
+        run_id: Uuid,
+    ) -> Result<Self, diesel::result::Error> {
+        let run_subquery = run::dsl::run
+            .filter(run::dsl::run_id.eq(run_id))
+            .select(run::dsl::run_group_id);
+        run_group_is_from_github
+            .filter(run_group_id.nullable().eq_any(run_subquery))
+            .first::<Self>(conn)
+    }
+
+    /// Queries the DB for run_group_is_from_githubs matching the specified query criteria
+    ///
+    /// Queries the DB using `conn` to retrieve run_group_is_from_githubs matching the crieria in `params`
+    /// Returns a result containing either a vector of the retrieved run_group_is_from_githubs as
+    /// RunGroupIsFromGithubData instances or an error if the query fails for some reason
+    ///
+    /// This is function is currently not in use, but it's functionality will likely be necessary in
+    /// the future, so it is included
+    #[allow(dead_code)]
+    pub fn find(
+        conn: &PgConnection,
+        params: RunGroupIsFromGithubQuery,
+    ) -> Result<Vec<Self>, diesel::result::Error> {
+        // Put the query into a box (pointer) so it can be built dynamically
+        let mut query = run_group_is_from_github.into_boxed();
+
+        // Add filters for each of the other params if they have values
```

Comment:
There are a lot of these query functions lying around - is there a way to easily unify them?

Reply:
I've been thinking about this and the best way I can think of to unify these is by writing macros to generate them.  I was initially averse to taking that approach because of my lack of familiarity with macros and also I couldn't figure out a good way to organize them while still allowing custom logic in the place where it's necessary.  I have an approach in mind now for how I'd do it.  I'd just need to spend some time messing around with them to see if I can actually make it work the way I want.  I think this makes the most sense as a thing I spend like a week trying to do, and, if I can't figure it out in that time, drop it until later.  Once it's implemented, it would for sure save time; I just don't know how long it would take to figure it out.  Thoughts?

Reply:
I think that approach makes sense.  I don't think you need to worry about this for now since this is all coded up already.  Maybe after the 1.0 release it can happen as part of a series of larger refactors / updates (hopefully with some help).  

Add an issue / code todo to track this for now so it doesn't get lost in the shuffle.